Fixed non-list tolist() bug

rgerkin opened the pull request:
Fixes #185.

```
import sys
sys.path.insert(0, '/home/rgerkin/Dropbox/dev/quantities')
import numpy as np
import quantities as pq

x = np.array([1.0, 2.0])
y = np.array(3.0)

x.tolist()  # Works fine
y.tolist() # Works fine

(x * pq.V).tolist() # Works fine
(y * pq.V).tolist() # Throws exception (fixed by this PR).
```

thisch commented:
Thx for this fix. Please also add a unit-test for it somewhere here https://github.com/python-quantities/python-quantities/blob/6ee561e918c00fec773481fa9a2c73e8e106f061/quantities/tests/test_methods.py#L13

rgerkin commented:
I've updated the unit test to check this.  An expected failure also succeeds now, although I think this might be due to a change (for the good) in numpy behavior, so I updated that test as well.

rgerkin commented:
@thisch Is there anything else you need from me or can this be merged and a new release created?  Travis passes for all non-archaic numpy versions.

thisch commented on quantities/tests/test_umath.py:
```diff
@@ -154,17 +154,13 @@ def test_ceil(self):
             [-1., -1., -0., 1., 2., 2., 2.] * pq.m
             )
 
-    @unittest.expectedFailure
```
Why was this change and the change below needed? It doesn't seem to be related to the bug fix in the tolist mehtod.

rgerkin replied:
@thisch The numpy `.fix` method is something that we would like to behave the same in numpy and in quantities (i.e. it should round towards zero).  It used to not work with quantities, which is why the expected failure was there.  But it works now, because of changes in numpy I think (though it could be the result of my changes)?  So I removed the expected failure because it now passes as it should.

thisch commented:
> I've updated the unit test to check this. An expected failure also succeeds now, although I think this might be due to a change (for the good) in numpy behavior, so I updated that test as well.

I see.

Thx for the fix and the unittest.  IMO this is ready for being merged.

apdavison commented:
I'd like to update the test matrix to remove older versions of NumPy, but we need to migrate to travis-ci.com first (see #187)
